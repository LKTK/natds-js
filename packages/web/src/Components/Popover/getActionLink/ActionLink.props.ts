import { ILinkProps } from "../../Link";

export interface IActionLink extends Omit<ILinkProps, "children" | "theme"> {

  /**
   * Please use `text` instead of `children`.
   *
   * @optional
   * @todo adopt `children` as primary property instead of `text`
   */
  children?: ILinkProps["children"];

  href?: ILinkProps["href"];

  onClick?: ILinkProps["onClick"];

  /**
   * @required
   * @todo deprecate `text` property in favor of `children`
   */
  text: string;
}
