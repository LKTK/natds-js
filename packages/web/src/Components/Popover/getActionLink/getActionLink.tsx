import * as React from "react";
import Link from "../../Link";
import { IActionLink } from "./ActionLink.props";

export const getActionLink: (actionLink?: IActionLink) => (JSX.Element | null) = (actionLink?: IActionLink) => {

  if (!actionLink) {
    return null;
  }

  const {
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    children,
    text,
    ...otherProps
  } = actionLink;

  return (
    <Link {...otherProps} variant="body2" className="popover-action-link">
      {text}
    </Link>
  );

};

export default getActionLink;
