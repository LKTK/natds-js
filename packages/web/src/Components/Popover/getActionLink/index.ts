import { getActionLink } from "./getActionLink";

export { getActionLink } from "./getActionLink";
export default getActionLink;
