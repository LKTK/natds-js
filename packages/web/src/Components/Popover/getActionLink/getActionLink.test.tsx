import * as TestRenderer from "react-test-renderer";
import getActionLink from "./getActionLink";

describe("Action link component", () => {
  describe("when no props are provided", () => {
    it("should match snapshot", () => {
      const testRenderer : TestRenderer.ReactTestRenderer = TestRenderer.create(getActionLink());

      expect(testRenderer).toMatchSnapshot();
    });
  });
  describe("when text property is provided", () => {
    it("should render correctly", () => {
      const testRenderer : TestRenderer.ReactTestRenderer = TestRenderer.create(getActionLink({
        text: "Test text",
      }));

      expect(testRenderer).toMatchSnapshot();
    });
  });
  describe("when both text and children properties are provided", () => {
    it("should render correctly", () => {
      const testRenderer : TestRenderer.ReactTestRenderer = TestRenderer.create(getActionLink({
        children: "Test children content",
        text: "Test text",
      }));

      expect(testRenderer).toMatchSnapshot();
    });
  });
});
