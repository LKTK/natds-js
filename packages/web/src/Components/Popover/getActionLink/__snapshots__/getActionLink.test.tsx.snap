// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action link component when both text and children properties are provided should render correctly 1`] = `
<a
  className="MuiTypography-root MuiLink-root MuiLink-underlineHover NatDSLink-default-1 NatDSLink-default-3 popover-action-link MuiTypography-body2 MuiTypography-colorInherit"
  onBlur={[Function]}
  onFocus={[Function]}
>
  Test text
</a>
`;

exports[`Action link component when no props are provided should match snapshot 1`] = `null`;

exports[`Action link component when text property is provided should render correctly 1`] = `
<a
  className="MuiTypography-root MuiLink-root MuiLink-underlineHover NatDSLink-default-1 NatDSLink-default-2 popover-action-link MuiTypography-body2 MuiTypography-colorInherit"
  onBlur={[Function]}
  onFocus={[Function]}
>
  Test text
</a>
`;
